pkg/agent/multicluster/policy_only_route_controller.go

```diff
@@ -0,0 +1,306 @@
+	if c.activeGWName == "" || c.nodeConfig.Name != c.activeGWName {
+		klog.InfoS("No active gateway or the Node is not the active Gateway", "node", c.nodeConfig.Name, "gateway", c.activeGWName)
+		return
+	}
+
+	podIP := pod.Status.PodIP
+	podNodeIP := pod.Status.HostIP
+	_, exists := c.installedPods[podIP]
+	if isDelete && exists {
+		c.queue.Add(&endpointChangeInfo{
+			isDelete: true,
+			podIP:    podIP,
+		})
+		delete(c.installedPods, podIP)
+		return
+	}
+
+	isValidPod := c.validatePod(pod)
+	if !isValidPod {
+		return
+	}
+	c.queue.Add(&endpointChangeInfo{
+		podNodeIP: podNodeIP,
+		podIP:     podIP,
+	})
+	c.installedPods[podIP] = podNodeIP
+}
+
+func (c *MCWithPolicyOnlyNodeRouteController) Run(stopCh <-chan struct{}) {
+	controllerName := "AntreaAgentMCWithPolicyOnlyNodeRouteController"
+	defer c.queue.ShutDown()
+
+	cacheSyncs := []cache.InformerSynced{c.podInformer.HasSynced, c.gwInformer.HasSynced}
+	klog.InfoS("Starting controller", "controller", controllerName)
+	defer klog.InfoS("Shutting down controller", "controller", controllerName)
+	if !cache.WaitForNamedCacheSync(controllerName, stopCh, cacheSyncs...) {
+		return
+	}
+
+	c.initialize()
+	for i := 0; i < 5; i++ {
+		go wait.Until(c.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (c *MCWithPolicyOnlyNodeRouteController) worker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *MCWithPolicyOnlyNodeRouteController) processNextWorkItem() bool {
+	obj, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(obj)
+	if k, ok := obj.(*endpointChangeInfo); !ok {
+		c.queue.Forget(obj)
+		klog.InfoS("Expected endpointChangeInfo in work queue but got", "object", obj)
+		return true
+	} else if err := c.syncPodFlows(k); err == nil {
+		c.queue.Forget(k)
+	} else {
+		// Put the item back on the workqueue to handle any transient errors.
+		c.queue.AddRateLimited(k)
+		klog.ErrorS(err, "Error syncing key, requeuing", "key", k)
+	}
+	return true
+}
+
+func (c *MCWithPolicyOnlyNodeRouteController) initialize() {
+	activeGW, err := getActiveGateway(c.gwLister, c.namespace)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get an active Gateway")
+	}
+
+	if activeGW != nil {
+		c.mutex.Lock()
+		defer c.mutex.Unlock()
+		c.activeGWName = activeGW.Name
+		klog.InfoS("Found an active Gateway", "gateway", klog.KObj(activeGW))
+	}
+}
+
+func (c *MCWithPolicyOnlyNodeRouteController) syncPodFlows(key *endpointChangeInfo) error {
+	if key.isDelete {
+		klog.V(2).InfoS("Deleting Multi-cluster flows for Pod", "podIP", key.podIP)
+		if err := c.ofClient.UninstallMulticlusterFlows(fmt.Sprintf("pod_%s", key.podIP)); err != nil {
+			klog.ErrorS(err, "Failed to uninstall Multi-cluster flows for Pod", "podIP", key.podIP)
+			return err
+		}
+	} else {
+		klog.V(2).InfoS("Adding Multi-cluster flows for Pod", "podIP", key.podIP, "nodeIP", key.podNodeIP)
+		if err := c.ofClient.InstallMulticlusterPodFlows(net.ParseIP(key.podIP), net.ParseIP(key.podNodeIP)); err != nil {
```

Comment:
Better to also give the flow key when installing the flows to be consistent. The current implementation is the flow key is generated in pkg/agent/openflow with installation, and is used by route_controller when uninstallation. Or you can pass podIP in `UninstallMulticlusterFlows`, and let the openflow functions to deside the flow key?

Reply:
Done.